app/server/src/features/events/resolvers.ts

```diff
@@ -25,13 +31,80 @@ export const resolvers: Resolvers = {
+                const { id: eventId } = fromGlobalId(args.input.eventId);
+                const createBroadcastMessage = await Event.createBroadcastMessage(ctx.viewer.id, ctx.prisma, {
+                    ...args.input,
+                    eventId,
+                });
+                const formattedBroadcastMessage = toBroadcastMessageId(createBroadcastMessage);
+                const edge = {
+                    node: formattedBroadcastMessage,
+                    cursor: formattedBroadcastMessage.id,
+                };
+                ctx.pubsub.publish({
+                    topic: 'broadcastMessageCreated',
+                    payload: { eventId, broadcastMessageCreated: { edge } },
+                });
+                return edge;
+            });
+        },
+        async editBroadcastMessage(parent, args, ctx, info) {
+            return runMutation(async () => {
+                if (!ctx.viewer.id) throw new ProtectedError({ userMessage: errors.noLogin });
+                const { id: broadcastMessageId } = fromGlobalId(args.input.broadcastMessageId);
+                const editEventBroadcastMessage = await Event.editBroadcastMessage(
+                    broadcastMessageId,
+                    args.input.broadcastMessage,
+                    ctx.prisma
+                );
+                const formattedBroadcastMessage = toBroadcastMessageId(editEventBroadcastMessage);
+                const edge = {
+                    node: formattedBroadcastMessage,
+                    cursor: formattedBroadcastMessage.id,
+                };
+                ctx.pubsub.publish({
+                    topic: 'broadcastMessageEdited',
+                    payload: { eventId: editEventBroadcastMessage.eventId, broadcastMessageEdited: { edge } },
+                });
+                return edge;
+            });
+        },
+        async deleteBroadcastMessage(parent, args, ctx, info) {
+            return runMutation(async () => {
+                if (!ctx.viewer.id) throw new ProtectedError({ userMessage: errors.noLogin });
+                const { id: broadcastMessageId } = fromGlobalId(args.input.broadcastMessageId);
+                const deleteEventBroadcastMessage = await Event.updateBroadcastMessageVisibility(
+                    broadcastMessageId,
+                    args.input.toggleBroadcastMessageVisibility,
+                    ctx.prisma
+                );
+                const formattedBroadcastMessage = toBroadcastMessageId(deleteEventBroadcastMessage);
+                const edge = {
+                    node: formattedBroadcastMessage,
+                    cursor: formattedBroadcastMessage.id,
+                };
+                ctx.pubsub.publish({
+                    topic: 'broadcastMessageDeleted',
+                    payload: { eventId: deleteEventBroadcastMessage.eventId, broadcastMessageDeleted: { edge } },
+                });
+                return edge;
+            });
+        },
```

Comment:
I think I mentioned this already, but these should be in separate resolvers under their own folder rather than in events.

Reply:
done